src/main/kotlin/me/partlysanestudios/partlysaneskies/features/chat/ChatTransformer.kt

```diff
@@ -0,0 +1,49 @@
+package me.partlysanestudios.partlysaneskies.features.chat
+
+import cc.polyfrost.oneconfig.events.event.ChatSendEvent
+import cc.polyfrost.oneconfig.libs.eventbus.Subscribe
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.config
+import net.minecraft.client.Minecraft
+
+
+//Currently only using to owoify send chat messages (meow)
+object ChatTransformer {
+    var lastmsg = ""
+    @Subscribe
+    fun onChat(event: ChatSendEvent){
+        //dont mess with chats that dont want to
+             if(!doChatTransform()) return
+
+            event.isCancelled = true
+            val msg = event.message
+        //Do not go into recursion (bad)
+            if(lastmsg == msg) {
+                event.isCancelled = false
+                return
+            }
+            //dont break commands :)
+            if(msg.startsWith("/")) return
+
+            val player = Minecraft.getMinecraft().thePlayer
+            var transformedmsg: String? = null
+
+            if(config.transformOWO){
+                transformedmsg = OwO.owoify(msg)
+            }
+
+            if(transformedmsg == null) return
+        // I dont know why putting this in a thread fixes it, but I do not complain
+        Thread{
+            lastmsg = transformedmsg
```

Comment:
I dont see why you need this, but im sure its some funny thread thing
Please explain

Reply:
The thread is needed, because otherwise you would either have no send messages or send two identical messages due to the player.sendChatMessage() calling the onChat() function recursively

Reply:
Im not talking about the thread itself, im talking about assigning that var and then reassigning it immediately

Reply:
the  if(transformedmsg == null) return is just a null failsafe

the lastmsg functions as a recursion anchor wich after the recursive call gets stopped in
 if (lastmsg == msg) {
            event.isCancelled = false
            return
        }

it needs to be resetted so you are able to send the same message twice